File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> s;
        vector<int> res;
        for(int i=0;i<nums.size();i++){
            if(s.find(target-nums[i])!=s.end()){
                res.push_back(s[target-nums[i]]);
                res.push_back(i);
            }else{
                s.insert({nums[i],i});
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1
        while l<=r:
            mid = l + (int((r-l)/2))
            # print(mid)
            if target>nums[mid]:
                l = mid+1
            elif target == nums[mid]:
                return mid
            else:
                r = mid-1
        return -1
        